addons/binding/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/ApiserviceHandler.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /**
+     * Request auth token with read or write access.
+     * (Write access is prepared for a possible later usage for updating meters.)
+     *
+     * @param user     The username to use
+     * @param password The corresponding password
+     * @param scope    The granted scope on the api for the binding
+     */
+    private void obtainAuthTokenAndExpiryDate(String user, String password, String scope) {
+        try {
+
+            String url = new StringBuilder("https://pixometer.io/api/v1/access-token/").toString();
+            Properties urlHeader = (Properties) new Properties().put("CONTENT-TYPE", "application/json");
+
+            JsonObject httpBody = new JsonObject();
+            httpBody.addProperty("username", user);
+            httpBody.addProperty("password", password);
+            httpBody.addProperty("scope", scope);
+
+            InputStream content = new ByteArrayInputStream(httpBody.toString().getBytes(StandardCharsets.UTF_8));
+            String urlResponse = HttpUtil.executeUrl("POST", url, urlHeader, content, "application/json", 2000);
+            JsonObject responseJson = (JsonObject) new JsonParser().parse(urlResponse);
+
+            if (responseJson.has(CONFIG_BRIDGE_AUTH_TOKEN)) {
+
+                // Store the expire date for automatic token refresh
+                Integer expiresIn = Integer.parseInt(responseJson.get("expires_in").toString());
+                setTokenExpiryDate(ZonedDateTime.now().plusSeconds(expiresIn));
+
+                setAuthToken(responseJson.get(CONFIG_BRIDGE_AUTH_TOKEN).toString().replaceAll("\"", ""));
+                updateStatus(ThingStatus.ONLINE);
+                return;
+            }
+
+            String errorMsg = new StringBuilder("Invalid Api Response ( ").append(responseJson.toString()).append(" )")
+                    .toString();
+
+            throw new Exception(errorMsg);
+
+        } catch (Exception e) {
+            String errorMsg = new StringBuilder("Could not obtain auth token.").append(" ")
+                    .append("Please check your configured account credentials.")
+                    .append(System.getProperty("line.separator")).append(this.getThing().getUID())
+                    .append(System.getProperty("line.separator")).append(e.getMessage()).toString();
+
+            logger.error(errorMsg, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+        }
+    }
+
+    /**
+     * Getters and Setters
+     */
+
+    public String getAuthToken() {
+        return authToken;
+    }
+
+    private void setAuthToken(String authToken) {
+        this.authToken = authToken;
+    }
+
+    public int getRefreshInterval() {
+        return refreshInterval;
+    }
+
+    private void setRefreshInterval(int refreshInterval) {
+        this.refreshInterval = refreshInterval;
+    }
+
+    public ZonedDateTime getTokenExpiryDate() {
```

Comment:
To keep it simple I would use a long instead of a more complex `ZonedDateTime` also because the value it is based on `expires_in` is just a relative number (of seconds)  in combination with `System.nanoTime()` and `TimeUnit.SECONDS.toNanos(expires_in)` You even might want to have a look at `ExpiringCache`